packages/consumption/src/modules/requests/itemProcessors/readAttribute/ReadAttributeRequestItemProcessor.ts

```diff
@@ -1,49 +1,172 @@
             }
 
-            const ownerIsCurrentIdentity = this.accountController.identity.isMe(foundAttribute.content.owner);
-            if (!ownerIsCurrentIdentity && foundAttribute.content instanceof IdentityAttribute) {
-                return ValidationResult.error(CoreErrors.requests.invalidRequestItem("The given Attribute belongs to someone else. You can only share own Attributes."));
+            attribute = foundLocalAttribute.content;
+
+            if (_requestItem.query instanceof IdentityAttributeQuery && attribute instanceof IdentityAttribute && this.accountController.identity.isMe(attribute.owner)) {
+                if (foundLocalAttribute.isShared()) {
+                    return ValidationResult.error(
+                        CoreErrors.requests.attributeQueryMismatch(
+                            "The provided IdentityAttribute is a shared copy of a RepositoryAttribute. You can only share RepositoryAttributes."
+                        )
+                    );
+                }
+
+                const ownSharedIdentityAttributeVersions = await this.consumptionController.attributes.getSharedVersionsOfRepositoryAttribute(
+                    foundLocalAttribute.id,
+                    [requestInfo.peer],
+                    false
+                );
+                const sourceAttributeIdsOfOwnSharedIdentityAttributeVersions = ownSharedIdentityAttributeVersions.map((ownSharedIdentityAttribute) =>
+                    ownSharedIdentityAttribute.shareInfo?.sourceAttribute?.toString()
+                );
+
+                let repositoryAttribute = foundLocalAttribute;
+                let i = 0;
+                while (repositoryAttribute.succeededBy !== undefined && i < 1000) {
+                    const successor = await this.consumptionController.attributes.getLocalAttribute(repositoryAttribute.succeededBy);
+                    if (!successor) {
+                        throw TransportCoreErrors.general.recordNotFound(LocalAttribute, repositoryAttribute.succeededBy.toString());
+                    }
+                    if (sourceAttributeIdsOfOwnSharedIdentityAttributeVersions.includes(successor.id.toString())) {
+                        return ValidationResult.error(
+                            CoreErrors.requests.attributeQueryMismatch(
+                                `The provided IdentityAttribute is outdated. You have already shared the Successor '${successor.id.toString()}' of it.`
+                            )
+                        );
+                    }
+                    repositoryAttribute = successor;
+                    i++;
+                }
+            }
+
+            if (_requestItem.query instanceof ThirdPartyRelationshipAttributeQuery && attribute instanceof RelationshipAttribute) {
+                if (!foundLocalAttribute.isShared()) {
+                    throw new Error("this should never happen");
```

Comment:
"This should never happen because you can only create a RelationshipAttribute involving another peer and therefore it will be automatically shared."

This would be better?

Reply:
I'd rather be in favor of "The LocalAttribute found is faulty because its `shareInfo` property is undefined, although its `content` is given by a RelationshipAttribute. Since RelationshipAttributes only make sense in the context of Relationships, they must always be shared." or something like that. I'll rewrite this somehow later, but perhaps it would be more elegant overall if we prevent LocalAttributes with RelationshipAttribute as `content` and undefined `shareInfo` to be exist in the first place. But I don't know if this can be done in such a way that there is no need for a `"this should never happen"` kind of error message here, because this error should indeed never be thrown. (But without the error, it is complained that the property `shareInfo` might not be set.)